### PR TITLE
Use class

### PR DESCRIPTION
**Pull request by bakura10**

### CODE REVIEW

File: src/Container/ApplicationFactory.php [hunk 1]
```diff
@@ -119,16 +122,16 @@ class ApplicationFactory
...
     {
-        $router = $services->has('Zend\Expressive\Router\RouterInterface')
-            ? $services->get('Zend\Expressive\Router\RouterInterface')
+        $router = $services->has(RouterInterface::class)
+            ? $services->get(RouterInterface::class)
             : new AuraRouter();
 
-        $finalHandler = $services->has('Zend\Expressive\FinalHandler')
-            ? $services->get('Zend\Expressive\FinalHandler')
+        $finalHandler = $services->has(FinalHandler::class)
```
Comment (weierophinney) [inline src/Container/ApplicationFactory.php]: In this case, no; the final handler does not need to be an instance of `Zend\Stratigility\FinalHandler`; it can be a simple `callable`. This is the reason why the tests fail currently.

The reason I used `Zend\Expressive` for the namespace is because we have some plans to offer an alternate final handler solution that will provide templated error pages. However, again, we wouldn't want to typehint on that class, as any callable is allowed in the signature that consumes the service.

In other words, it's a service name that doesn't map 1:1 with a class.

**Comment by bakura10** [comment]
Oops... it seems like a good reason to use `Zend\Stratigility\FinalHandler` instead of `Zend\Expressive\FinalHandler` ;).

**Comment by weierophinney** [comment]
@bakura10 You'll need to rebase; I've merged most of your PRs at this point!

**Comment by bakura10** [comment]
grrr, seems I still suck at rebasing...

**Comment by samsonasik** [comment]
@bakura10 you can do following steps:
- git checkout master && git pull git://github.com/zendframework/zend-expressive.git master
- git checkout class && git rebase master

after that, if you have conflict, you can do:
- fix conflict <<<HEAD
- don't add new commit
- git add .
- git rebase --continue

And last, do:
- git push --force origin class

**Comment by weierophinney** [comment]
@bakura10 I took care of the conflicts and got it merged. Thanks!
